### ISSUE TITLE
assocPath seems to be broken in 0.23.0

### ISSUE THREAD

**Issue by pavel-dohnal-momentumft** [ramda/ramda#2040]
This is my code:
```
const scope = {
  baseValues: undefined,
}

assocPath(
  ["baseValues", "income", "categoryGroups"],
  { '9': { categories: { '32': 4 } } },
  scope,
)
```

Which works in 0.22.1 but is broken in 0.23.0. The error is: "Cannot convert undefined or null to object". There is nothing mentioned in the upgrade guide, so I assume this is not expected behaviour.

**Comment by CrossEye** [comment]
There was a significant update to this function in #1926.  I assume that's where this changed  I have updated the upgrade guide from

> - #1926 Integer path segments are now accepted in `R.path` and related functions

to 

> - #1926 Integer path segments are now accepted in `R.path` and related functions `R.assocPath`, `R.lensPath`, `R.pathEq`, `R.pathOr`, and `R.pathSatisfies`

This regression was not intentional.  I don't think it's a case we ever considered.  It works on `const scope = {}`

It seems a simple fix.  I think it's worth doing.  I will create a PR.

**Comment by Tom910** [comment]
When released the new version?

**Comment by CrossEye** [comment]
We haven't yet.  I will try to get a patch release out next weekend.

**Comment by burt202** [comment]
I think Ive found another regression in `assocPath` in v0.23. The following code seems to work fine in v0.22.

```
const num1 = 2
const num2 = 2

const data = {
  aaa: {
    bbb: {}
  }
}

assocPath(
  ["aaa", "bbb", num1, "ccc", num2],
  {value: 333},
  data
)
```

returns...

```{"aaa": {"bbb": {"2": {"ccc": [undefined, undefined, {"value": 333}]}}}}```

instead of...

```{"aaa": {"bbb": {"2": {"ccc": {"2": {"value": 333}}}}}}```

**Comment by CrossEye** [comment]
Closed in favor of #2315.
